File: eBiblioteka/UI/admin_ebiblioteka/lib/models/rating.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/book.dart';
import '../models/user.dart';

part 'rating.g.dart';

@JsonSerializable()
class Rating {
  int? id;
  int? stars;
  String? comment;
  DateTime? dateTime;
  int? userId;
  User? user;
  int? bookId;
  Book? book;

  Rating(this.id, this.stars, this.comment, this.dateTime, this.bookId,
      this.userId, this.user, this.book);

  factory Rating.fromJson(Map<String, dynamic> json) => _$RatingFromJson(json);

  Map<String, dynamic> toJson() => _$RatingToJson(this);
}

File: eBiblioteka/eBiblioteka.Core/Entities/Rating.cs

namespace eBiblioteka.Core
{
    public class Rating:BaseEntity
    {
        public int Stars { get; set; }
        public string? Comment { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public int BookId { get; set; }
        public Book Book { get; set; } = null!;
    }
}
